dev/com.ibm.ws.webcontainer.servlet.4.0/src/com/ibm/ws/webcontainer40/srt/SRTServletRequest40.java

```diff
@@ -194,8 +194,13 @@ public HttpServletMapping getCurrentHttpServletMapping(WebAppDispatcherContext40
                 case PATH:
                     // in the case of /* mapping, skip the pathInfo.substring.
                     if (!dispatchContext.hasSlashStarMapping()) {
-                        // matchValue is the pathInfo after the last "/" and pattern is the servletPath + "/*"
-                        matchValue = pathInfo.substring(pathInfo.lastIndexOf("/") + 1);
+                        if (com.ibm.ws.webcontainer.osgi.WebContainer.getServletContainerSpecLevel() >= com.ibm.ws.webcontainer.osgi.WebContainer.SPEC_LEVEL_61) {
```

Comment:
Doc this behavior change in main issue

https://github.com/OpenLiberty/open-liberty/issues/27383